C/services/common/notification_queue.cpp

```diff
@@ -676,15 +676,14 @@ void NotificationQueue::processAllDataBuffers(const string& assetName)
 {
 	// Get the subscriptions instance
 	NotificationSubscription* subscriptions = NotificationSubscription::getInstance();
-	// Get all subscriptions for assetName
-	subscriptions->lockSubscriptions();
-	std::vector<SubscriptionElement>&
-		registeredItems = subscriptions->getSubscription(assetName);
-	subscriptions->unlockSubscriptions();
 	if (!subscriptions)
 	{
 		return;
 	}
+	// Get all subscriptions for assetName
+	subscriptions->lockSubscriptions();
```

Comment:
You need to be careful as whistle holding this lock you also take out the lock on the instances. Make sure you always do these two locks in the same order to avoid deadlocks.